main.py

```diff
@@ -40,311 +40,135 @@
 from inferno import *
 from scorescreen import *
 from howtoplay import *
+from button import Button
 
+disp_width = 600
+disp_height = 600
+
+score_path = os.path.join(get_activity_root(), 'data', 'score.pkl')
+
+black = (0, 0, 0)
+white = (255, 255, 255)
 
 class game:
 
     sound = True
 
-    def run(self):
-
+    def __init__(self):
         self.crashed = False
         self.running_mode = None
-        black = (0, 0, 0)
-        white = (255, 255, 255)
-        clock = pygame.time.Clock()
+        self.clock = pygame.time.Clock()
         timer = pygame.time.Clock()
 
-        disp_width = 600
-        disp_height = 600
+        self.gameDisplay = None
+        self.info = None
+
+        self.buttons = []
+
+        self.maxscore = [0, 0, 0, 0, 0, 0]
+    
+    def run_game(self, window, gamenumber):
+        self.running_mode = window()
+        self.running_mode.crashed = self.crashed
+        s = self.running_mode.run(self.gameDisplay, self.info)
 
-        press = 0
+        if scorewindow(self.gameDisplay, s, gamenumber).run():
```

Comment:
How come you're passing args to `scorewindow` when `scorewindow.run` requires the args and not `scorewindow`?

Reply:
Seen your refactor.